### PR TITLE
Fix plugins_url() when used inside a theme

### PR DESCRIPTION
**Pull request by WPprodigy**
On WPcom, it's common to have plugins located within the theme folder. As a result, some might have become dependent on `plugins_url()` which would have resolved fine on WPcom.

On VIP Go though, `wp-content/themes/theme-name/plugins` is not a valid entry in `$wp_plugin_paths`. So `plugin_basename()` ends up returning a rather funky path, which results in URLs that are both invalid and leaking server paths.

### CODE REVIEW

File: wpcom-plugins.php [hunk 1]
```diff
@@ -22,6 +22,35 @@ function wpcom_vip_legacy_load_plugin( $plugin = false, $folder = false, $versio
...
+ * Setting a high priority (100) to allow clients to resolve this problem before our fix belows needs to get involved.
+ *
+ * @param string $url    The complete URL to the plugins directory including scheme and path.
+ * @param string $path   Path relative to the URL to the plugins directory. Blank string if no path is specified.
+ * @param string $plugin The plugin file path to be relative to. Blank string if no plugin is specified.
+ *
+ * @return $url
+ */
+function vip_wpcom_compat_fix_theme_relative_plugins_url( $url, $path, $plugin ) {
+	if ( empty( $plugin ) || ! is_string( $plugin ) || false === strpos( $url, '/var/www/wp-content/themes/' ) ) {
```
Comment (rinatkhaziev) [inline wpcom-plugins.php]: What's the condition where `$plugin` might not be a string?

Comment (WPprodigy) [reply wpcom-plugins.php]: Basically would just be a user error. I just noticed that core checks for it here: https://developer.wordpress.org/reference/functions/plugins_url/

Though it does seem safe to assume we are getting a string, even if empty. Notably there is no check here already: https://github.com/Automattic/vip-go-mu-plugins/blob/d1361162e2436f81bcc7493a16af073dd4cb3914/z-client-mu-plugins.php#L84-L113, so perhaps we can just remove in this fix as well.

**Comment by WPprodigy** [comment]
@rinatkhaziev I updated this PR. 

In summary, if the `$relative_file_path` points to something in the theme's directory then we know we need to intervene and resolve the problem, much like what we do for client-mu-plugins: https://github.com/Automattic/vip-go-mu-plugins/blob/d1361162e2436f81bcc7493a16af073dd4cb3914/z-client-mu-plugins.php#L84-L113

For testing, I just sandboxed a site and put the following in the theme's functions.php file:

```
add_action( 'init', 'testing_plugins_url_stuff' );
function testing_plugins_url_stuff() {
    echo ' Test 1: ' . plugins_url( '', __FILE__ );
    echo ' Test 2: ' . plugins_url( 'file.js', __FILE__ );
    echo ' Test 3: ' . plugins_url( 'some/path/to/file.js', __FILE__ );
    die();
}
```

Before this patch, the above will output really messed up URLs that disclose full server file paths. After the patch, they will be "valid" urls as intended by `plugins_url()` usage.

Also fixes problems with `plugin_dir_url()` by inheritance.

**Review by rinatkhaziev — APPROVED** [review]
LGTM!

**Pull request merged**